Documentation/install/system_requirements.rst

```diff
@@ -104,6 +104,41 @@ linked, either choice is valid.
    Users running Linux 4.10 or earlier with Cilium CIDR policies may face
    :ref:`cidr_limitations`.
 
+L7 proxy redirection currently uses ``TPROXY`` iptables actions as well
+as ``socket`` matches. For L7 redirection to work as intended kernel
+configuration must include the following modules:
+
+.. code:: bash
+
+        CONFIG_NETFILTER_XT_TARGET_TPROXY=m
+        CONFIG_NETFILTER_XT_MATCH_MARK=m
+        CONFIG_NETFILTER_XT_MATCH_SOCKET=m
+
+.. note::
+
+   As noted above, ``xt_socket`` kernel module is generally needed for
+   L7 redirection. Specifically, when ``xt_socket`` kernel module is
+   missing the forwarding of redirected L7 traffic does not work
+   in non-tunneled datapath modes. Since some notable kernels (e.g.,
+   COS) are shipping without ``xt_socket`` module, Cilium implements a
+   fallback compatibility mode to allow L7 policies and
+   visibility to be used with those kernels. Currently this fallback
+   disables ``ip_early_demux`` kernel feature in non-tunneled datapath
+   modes, which may decrease system networking performance.
+
+If ``xt_socket`` is missing and Cilium is run in a non-tunneled datapath
+mode ``ip_early_demux`` system control setting is disabled by default.
```

Comment:
This sentence seems to be duplicated with the text in the current note, do we need to say it twice?

Reply:
paragraphs combined.